src/state/lottery/getUserLotteryData.ts

```diff
@@ -0,0 +1,147 @@
+import { request, gql } from 'graphql-request'
+import { GRAPH_API_LOTTERY } from 'config/constants/endpoints'
+import { LotteryTicket } from 'config/constants/types'
+import { LotteryUserGraphEntity, LotteryResponse, UserRound } from 'state/types'
+import { getRoundIdsArray, fetchMultipleLotteries, hasRoundBeenClaimed, processRawTicketsResponse } from './helpers'
+import { fetchUserTicketsForMultipleRounds } from './fetchUnclaimedUserRewards'
+
+const applyNodeDataToUserGraphResponse = (
+  nodeData: LotteryResponse[],
+  graphData: UserRound[],
+  ticketData: { roundId: string; userTickets: LotteryTicket[] }[],
+): UserRound[] => {
+  //   If no graph rounds response - return node data
+  if (graphData.length === 0) {
+    return nodeData.map((nodeRound) => {
+      const ticketDataForRound = ticketData.find((roundTickets) => roundTickets.roundId === nodeRound.lotteryId)
+      return {
+        endTime: nodeRound.endTime,
+        status: nodeRound.status,
+        lotteryId: nodeRound.lotteryId.toString(),
+        claimed: hasRoundBeenClaimed(ticketDataForRound.userTickets),
+        totalTickets: null,
+      }
+    })
+  }
+
+  //   Else if there is a graph response - merge with node data where node data is more accurate
+  const mergedResponse = graphData.map((graphRound, index) => {
+    const nodeRound = nodeData[index]
+    // if there is node data for this index, overwrite graph data. Otherwise - return graph data.
+    if (nodeRound) {
+      const ticketDataForRound = ticketData.find((roundTickets) => roundTickets.roundId === nodeRound.lotteryId)
+      // if isLoading === true, there has been a node error - return graphRound
+      if (!nodeRound.isLoading) {
+        return {
+          endTime: nodeRound.endTime,
+          status: nodeRound.status,
+          lotteryId: nodeRound.lotteryId.toString(),
+          claimed: hasRoundBeenClaimed(ticketDataForRound.userTickets),
+          totalTickets: graphRound.totalTickets,
+        }
+      }
+      return graphRound
+    }
+    return graphRound
+  })
+  return mergedResponse
+}
+
+const getUserLotteryData = async (account: string, currentLotteryId: string): Promise<LotteryUserGraphEntity> => {
```

Comment:
New user data fetch - called by `fetchUserTicketsAndLotteries`, fired after ticket purchases and on load/account change

& `fetchUserLotteries`, fires as rounds transition and after successful user reward claims